src/types.rs

```diff
@@ -136,3 +136,136 @@ impl LintPass for UnitCmp {
         }
     }
 }
+
+pub struct CastPass;
+
+declare_lint!(pub CAST_PRECISION_LOSS, Allow,
+              "casts that cause loss of precision");
+declare_lint!(pub CAST_SIGN_LOSS, Allow,
+              "casts from signed types to unsigned types");
+declare_lint!(pub CAST_POSSIBLE_OVERFLOW, Allow,
+              "casts that may cause overflow");
```

Comment:
+1, I was suggesting the same thing for the outputted error message. These errors can be subtle, esp the first one.